### ISSUE TITLE
Docker: user cannot write in working directory

### ISSUE THREAD

**Issue by hustlahusky** [mikefarah/yq#1014]
**Describe the bug**
I am using docker image to build json file from yaml for my swagger ui container. My Dockerfile:

```
FROM mikefarah/yq:4 as builder

COPY ./resources/apidoc ./
RUN yq eval -o=json -I=0 v1.yml > v1.json

FROM swaggerapi/swagger-ui:v3.40.0

ENV SWAGGER_JSON='/apidoc/v1.json'

COPY --from=builder /workdir /apidoc
```

Build failed on step 3 with permission denied error:

```
Step 1/6 : FROM mikefarah/yq:4 as builder
4: Pulling from mikefarah/yq
97518928ae5f: Already exists
58082ff9f0e7: Pull complete
5281cfdcecfd: Pull complete
d09e5d4afbea: Pull complete
Digest: sha256:07ba177e02af2232a41286f656d73a4924856d4a6b4635450a8f381bbc42af44
Status: Downloaded newer image for mikefarah/yq:4
 ---> 466fed4bbbe7
Step 2/6 : COPY ./resources/apidoc ./
 ---> 9ff17f86ded5
Step 3/6 : RUN yq eval -o=json -I=0 v1.yml > v1.json
 ---> Running in e60f665e888a
/bin/sh: can't create v1.json: Permission denied
```

It seems this error caused by latest release when WORKDIR defined before USER

https://github.com/mikefarah/yq/blob/183007d120403687f023aaf2c105c304e1e28422/Dockerfile#L18-L23

Version of yq: 4.15.1
Operating system: linux
Installed via: docker

**Comment by mikefarah** [comment]
I would think you need to follow the instructions here: https://github.com/mikefarah/yq#running-as-root

Not why moving that WORKDIR would make a difference, however, if it previously worked I'm happy to rollback changes  - can you please confirm it was working in the mikefarah/yq:4.14.2 ?

**Comment by DuncanvR** [comment]
The easiest way to reproduce is running the image and touching a file. In v4.14.2, this succeeds (there is no output), whereas in v4.15.1, it throws an error:
```
> docker run --rm -it --entrypoint sh mikefarah/yq:4.14.2 -c 'touch a'
> docker run --rm -it --entrypoint sh mikefarah/yq:4.15.1 -c 'touch a'
touch: a: Permission denied
```

Looking at the owner of `/workdir` folder, that makes sense:
```
> docker run --rm -it --entrypoint sh mikefarah/yq:4.14.2 -c 'ls -ld /workdir'
drwxr-xr-x    2 yq       yq            4096 Nov 14 00:11 /workdir
> docker run --rm -it --entrypoint sh mikefarah/yq:4.15.1 -c 'ls -ld /workdir'
drwxr-xr-x    2 root     root          4096 Nov 24 00:02 /workdir
```

If you could change the owner of the `/workdir` folder back to `yq`, that would resolve this issue, I believe. Whether that's by moving the `WORKDIR` statement after `USER`, or adding a `RUN chown yq:yq /workdir`, doesn't really matter to me.

**Comment by mikefarah** [comment]
Great, thanks for making this very clear 👍🏼  - fix will be in the next release.

**Comment by mikefarah** [comment]
Fixed in 4.16.1:

```bash
 docker run --rm -it --entrypoint sh mikefarah/yq:4.16.1 -c 'ls -ld /workdir'
drwxr-xr-x    1 yq       yq            4096 Dec  4 03:02 /workdir

```
